Modules/_io/textio.c

```diff
@@ -1477,8 +1477,13 @@ textiowrapper_read_chunk(textio *self)
             Py_DECREF(next_input);
             goto fail;
         }
+        PyObject *snapshot = Py_BuildValue("NN", dec_flags, next_input);
```

Comment:
Declaring `snapshot` at the top of the block will likely fix the AppVeyor build.